project-base/app/config/packages/messenger.yaml

```diff
@@ -55,3 +55,10 @@ framework:
             Shopsys\FrameworkBundle\Model\Product\Recalculation\DispatchAllProductsMessage: product_recalculation_priority_regular
             Shopsys\FrameworkBundle\Model\Order\Messenger\PlacedOrderMessage: placed_order_transport
             Symfony\Component\Mailer\Messenger\SendEmailMessage: send_email_transport
+
+services:
+    Shopsys\FrameworkBundle\Component\Messenger\MessageBusDecorator:
+        decorates: 'messenger.bus.default'
+        arguments:
+            $transportDsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+            $messageBus: '@Shopsys\FrameworkBundle\Component\Messenger\MessageBusDecorator.inner'
```

Comment:
I know this was my idea originally, but now I have to question myself 😆 
Should be this service definition in the project-base? I like that it's immediately clear what's happening and that we use some decorator, but to have this working, some steps are necessary during upgrade.

If we decide to keep it in project-base, it will require the upgrade notes

Reply:
that is a good point. I moved the definition into the framework package to leverage the upgradable platform benefits